org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/handlers/BundleUtils.java

```
@@ -83,4 +120,19 @@ private static String getBundleLocation(String location, boolean useReference) t
 		}
 		return bundleLocation;
 	}
+
+	private static BundleInfo getBundleInfo(String bundleLocation) throws IOException {
+		JarFile jarFile = new JarFile(bundleLocation);
+		Manifest manifest = jarFile.getManifest();
+		Attributes mainAttributes = manifest.getMainAttributes();
+		String bundleVersion = mainAttributes.getValue(Constants.BUNDLE_VERSION);
+		String symbolicName = mainAttributes.getValue(Constants.BUNDLE_SYMBOLICNAME);
+		jarFile.close();
+
+		if (!StringUtils.isBlank(symbolicName) && symbolicName.indexOf(";") >= 0) {
+			symbolicName = symbolicName.substring(0, symbolicName.indexOf(";"));
+		}
+
+		return new BundleInfo(bundleVersion, symbolicName);
+	}

```

```java
private static BundleInfo getBundleInfo(String bundleLocation) throws IOException {
	try (JarFile jarFile = new JarFile(bundleLocation)){
		Manifest manifest = jarFile.getManifest();
		if (manifest != null) {
			Attributes mainAttributes = manifest.getMainAttributes();
			String bundleVersion = mainAttributes.getValue(Constants.BUNDLE_VERSION);
			String symbolicName = mainAttributes.getValue(Constants.BUNDLE_SYMBOLICNAME);
			if (StringUtils.isNotBlank(symbolicName) && symbolicName.indexOf(";") >= 0) {
				symbolicName = symbolicName.substring(0, symbolicName.indexOf(";"));
			}							
			return new BundleInfo(bundleVersion, symbolicName);
		}
	}
	return null;
}
```

---

alternatively, throw IOException instead of returning null